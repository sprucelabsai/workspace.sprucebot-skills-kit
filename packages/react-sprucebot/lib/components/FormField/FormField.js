'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.default = Field;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _reactTextareaAutosize = require('react-textarea-autosize');

var _reactTextareaAutosize2 = _interopRequireDefault(_reactTextareaAutosize);

var _Typography = require('../Typography/Typography');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Field(_ref) {
	var _ref$input = _ref.input,
	    value = _ref$input.value,
	    onChange = _ref$input.onChange,
	    name = _ref$input.name,
	    _ref$meta = _ref.meta,
	    touched = _ref$meta.touched,
	    error = _ref$meta.error,
	    type = _ref.type,
	    finePrint = _ref.finePrint,
	    label = _ref.label,
	    _ref$placeholder = _ref.placeholder,
	    placeholder = _ref$placeholder === undefined ? label : _ref$placeholder,
	    children = _ref.children;

	return _react2.default.createElement(
		'div',
		{ className: (0, _classnames2.default)('InputField', 'input__wrapper') },
		label && value && _react2.default.createElement(
			'span',
			{
				className: (0, _classnames2.default)('FieldLabel', 'input__mini__label', 'js-show-label')
			},
			label
		),
		_react2.default.cloneElement(children, {
			className: (0, _classnames2.default)({ with_label: !!label }),
			name: name,
			value: value,
			onChange: onChange,
			type: type,
			placeholder: placeholder
		}),
		touched && error && _react2.default.createElement(
			'span',
			{
				className: (0, _classnames2.default)('FieldError', 'input__error', 'error-is-visible')
			},
			error
		),
		finePrint && _react2.default.createElement(
			_Typography.Paragraph,
			{ fine: true },
			finePrint
		)
	);
}

Field.propTypes = {
	input: _propTypes2.default.shape({
		value: _propTypes2.default.any.isRequired,
		onChange: _propTypes2.default.func
	}).isRequired,
	meta: _propTypes2.default.shape({
		touched: _propTypes2.default.bool,
		error: _propTypes2.default.string
	}).isRequired,
	children: _propTypes2.default.element.isRequired
};