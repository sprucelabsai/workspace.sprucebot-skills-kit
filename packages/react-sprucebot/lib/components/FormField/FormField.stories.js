'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _FormField = require('./FormField');

var _FormField2 = _interopRequireDefault(_FormField);

var _FormField3 = require('./FormField.md');

var _FormField4 = _interopRequireDefault(_FormField3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('FormField', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_FormField4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_FormField2.default,
		{
			input: { value: '', onChange: function onChange() {
					return null;
				} },
			meta: { touched: false, error: null },
			type: 'text',
			label: 'Email Address',
			placeholder: 'hello@sprucebot.com'
		},
		_react2.default.createElement('input', {
			type: 'text',
			label: 'Email Address',
			placeholder: 'hello@sprucebot.com'
		})
	);
})));