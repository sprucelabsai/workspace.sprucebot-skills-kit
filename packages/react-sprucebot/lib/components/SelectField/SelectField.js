'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _FormField = require('../FormField/FormField');

var _FormField2 = _interopRequireDefault(_FormField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var TextAreaField = function TextAreaField(_ref) {
	var children = _ref.children,
	    props = _objectWithoutProperties(_ref, ['children']);

	return _react2.default.createElement(
		_FormField2.default,
		props,
		_react2.default.createElement(
			'select',
			_extends({}, props, {
				className: (0, _classnames2.default)('SelectField', 'custom_dropdown')
			}),
			children
		)
	);
};

exports.default = TextAreaField;