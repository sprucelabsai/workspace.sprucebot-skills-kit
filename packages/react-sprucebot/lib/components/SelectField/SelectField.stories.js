'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _SelectField = require('./SelectField');

var _SelectField2 = _interopRequireDefault(_SelectField);

var _SelectField3 = require('./SelectField.md');

var _SelectField4 = _interopRequireDefault(_SelectField3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('SelectField', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_SelectField4.default, function () {
	return _react2.default.createElement(_SelectField2.default, {
		input: { value: '', onChange: function onChange() {
				return null;
			} },
		meta: { touched: false, error: null }
	});
}));