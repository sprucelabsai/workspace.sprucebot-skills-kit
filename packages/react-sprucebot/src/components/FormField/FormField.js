import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import classnames from 'classnames'
import TextArea from 'react-textarea-autosize'

import { Paragraph as P } from '../Typography/Typography'

export const FieldWrapper = styled.div.attrs({
	className: classnames('InputField', 'input__wrapper')
})``

export const FieldError = styled.span.attrs({
	className: classnames('FieldError', 'input__error', 'error-is-visible')
})``

export const FieldLabel = styled.span.attrs({
	className: classnames('FieldLabel', 'input__mini__label', 'js-show-label')
})``

export default function Field({
	input: { value, onChange, name },
	meta: { touched, error },
	type,
	finePrint,
	label,
	placeholder = label,
	children
}) {
	return (
		<FieldWrapper>
			{label && value && <FieldLabel>{label}</FieldLabel>}
			{React.cloneElement(children, {
				className: classnames({ with_label: !!label }),
				name,
				value,
				onChange,
				type,
				placeholder
			})}
			{touched && error && <FieldError>{error}</FieldError>}
			{finePrint && <P fine>{finePrint}</P>}
		</FieldWrapper>
	)
}

Field.propTypes = {
	input: PropTypes.shape({
		value: PropTypes.any.isRequired,
		onChange: PropTypes.func.isRequired
	}).isRequired,
	meta: PropTypes.shape({
		touched: PropTypes.bool,
		error: PropTypes.string
	}).isRequired,
	children: PropTypes.element.isRequired
}
