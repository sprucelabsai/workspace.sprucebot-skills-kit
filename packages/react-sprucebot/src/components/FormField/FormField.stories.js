import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import FormField from './FormField'
import readme from './FormField.md'

const stories = storiesOf('FormField', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(
		readme,
		withInfo()(() => (
			<FormField
				input={{ value: '', onChange: () => null }}
				meta={{ touched: false, error: null }}
				type="text"
				label="Email Address"
				placeholder="hello@sprucebot.com"
			>
				<input
					type="text"
					label="Email Address"
					placeholder="hello@sprucebot.com"
				/>
			</FormField>
		))
	)
)
