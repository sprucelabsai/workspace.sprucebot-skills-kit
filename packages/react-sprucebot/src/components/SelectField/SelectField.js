import React from 'react'
import styled from 'styled-components'
import classnames from 'classnames'

import FormField from '../FormField/FormField'

const StyledSelect = styled.select.attrs({
	className: classnames('SelectField', 'custom_dropdown')
})``

export default function TextAreaField({ children, ...props }) {
	return (
		<FormField {...props}>
			<StyledSelect>{children}</StyledSelect>
		</FormField>
	)
}
