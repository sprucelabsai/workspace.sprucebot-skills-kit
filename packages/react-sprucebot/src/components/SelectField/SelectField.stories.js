import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import SelectField from './SelectField'
import readme from './SelectField.md'

const stories = storiesOf('SelectField', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(readme, () => (
		<SelectField
			input={{ value: '', onChange: () => null }}
			meta={{ touched: false, error: null }}
		/>
	))
)
