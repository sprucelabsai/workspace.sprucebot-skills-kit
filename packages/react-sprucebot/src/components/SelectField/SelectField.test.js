import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import SelectField from './SelectField'
import FormField from '../FormField/FormField'

test('it renders a FormField', () => {
	const wrapper = shallow(
		<SelectField input={{ value: 'test', onChange: jest.fn() }} meta={{}}>
			<option value="testval" />
		</SelectField>
	)

	expect(wrapper).toMatchSnapshot()

	expect(wrapper.find(FormField)).toHaveLength(1)
})
