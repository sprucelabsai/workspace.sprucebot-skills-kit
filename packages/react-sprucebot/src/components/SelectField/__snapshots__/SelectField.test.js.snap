// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField tests Should match the snapshot 1`] = `
<Field>
  <select
    className="SelectField custom_dropdown"
  />
</Field>
`;
